src/AddressRegistry.Producer.Snapshot.Oslo/ProducerProjections.cs

```diff
@@ -0,0 +1,304 @@
+namespace AddressRegistry.Producer.Snapshot.Oslo
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using AddressRegistry.StreetName.Events;
+    using Be.Vlaanderen.Basisregisters.EventHandling;
+    using Be.Vlaanderen.Basisregisters.MessageHandling.Kafka.Simple;
+    using Be.Vlaanderen.Basisregisters.ProjectionHandling.Connector;
+    using Infrastructure;
+    using Microsoft.Extensions.Configuration;
+
+    [ConnectedProjectionName("Kafka producer")]
+    [ConnectedProjectionDescription("Projectie die berichten naar de kafka broker stuurt.")]
+    public sealed class ProducerProjections : ConnectedProjection<ProducerContext>
+    {
+        private readonly KafkaProducerOptions _kafkaOptions;
+        private readonly string _TopicKey = "Topic";
+
+        public ProducerProjections(IConfiguration configuration, ISnapshotManager snapshotManager)
+        {
+            var bootstrapServers = configuration["Kafka:BootstrapServers"];
+            var osloNamespace = configuration["OsloNamespace"];
+            osloNamespace = osloNamespace.TrimEnd('/');
+
+            var topic = $"{configuration[_TopicKey]}" ?? throw new ArgumentException($"Configuration has no value for {_TopicKey}");
+            _kafkaOptions = new KafkaProducerOptions(
+                bootstrapServers,
+                configuration["Kafka:SaslUserName"],
+                configuration["Kafka:SaslPassword"],
+                topic,
+                false,
+                EventsJsonSerializerSettingsProvider.CreateSerializerSettings());
+
+            When<Be.Vlaanderen.Basisregisters.ProjectionHandling.SqlStreamStore.Envelope<AddressBoxNumberWasCorrectedV2>>(async (_, message, ct) =>
```

Comment:
Can you order these is some sort of logical order. Like `proposed` -> `approved`, ... -> corrections -> changes